pkg/scheduler/scheduler.go

```diff
@@ -685,7 +685,9 @@ func (s *Scheduler) requeueAndUpdate(ctx context.Context, e entry) {
 
 	if e.status == notNominated || e.status == skipped {
 		patch := workload.AdmissionStatusPatch(e.Obj, true)
-		if workload.UnsetQuotaReservationWithCondition(patch, "Pending", e.inadmissibleMsg) {
+		change1 := workload.UnsetQuotaReservationWithCondition(patch, "Pending", e.inadmissibleMsg)
+		change2 := workload.PropagateResourceRequests(patch, &e.Info)
```

Comment:
```suggestion
		change1 := workload.UnsetQuotaReservationWithCondition(patch, "Pending", e.inadmissibleMsg)
		if features.Enabled(features.ConfigurableResourceTransformations) {
			change2 := workload.PropagateResourceRequests(patch, &e.Info)
		}
```

or

```suggestion
		change1 := workload.UnsetQuotaReservationWithCondition(patch, "Pending", e.inadmissibleMsg)
		if e.info.resourceTransformations != nil {
			change2 := workload.PropagateResourceRequests(patch, &e.Info)
		}
```

Could we guard this field change by feature gate?

Reply:
> Could we guard this field change by feature gate?

The status field is useful independent of the feature gate.   Even if the configurableResourceTransformation feature is disabled, the status field gives observability to other resource adjustments being done by Kueue (pod overheads, limit rangers, excluded resource prefixes).